andela/ah-infinity-stones
# #162947590  Fix like/dislike bugs

#### What does this PR do?
- Fix like and dislike bugs
#### Description of Task to be completed?
- Add likes and dislikes count fields
- add and update liking/disliking field on article payload
#### How should this be manually tested?
1. `git clone -b bg-fix-likesdislikes-bugs-162947590 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='ah_dev'
export DATABASE_USER='user'
export DATABASE_PASSWORD=pass'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'
export EMAIL_HOST_USER='username'
export EMAIL_HOST_PASSWORD='pass'
export SECRET_KEY='key-is-secret'
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
{ "user": {
"username": "username44",
"email": "username@user.com",
"password": "@Password321"	
}
}
```
Check your email and activate account. Use the registration token to acess the following endpoints:
- `http://127.0.0.1:8000/api/articles` - Post an article
- `http://127.0.0.1:8000/api/articles/<art_slug>/like-dislike` -  like or dislike the article created.
*Note* To know whether you like an article you must be logged in when fetching it.

8. Run tests`python3 manage.py test authors.apps.articles.tests.test_like_article`
#### What are the relevant pivotal tracker stories?
[#162947590](https://www.pivotaltracker.com/story/show/162947590)
#### Screenshots
<img width="965" alt="screenshot 2019-01-03 at 13 33 56" src="https://user-images.githubusercontent.com/12385482/50662442-7c477380-0fb7-11e9-98b6-b9aacbeed902.png">
